cmd/internal/kube/injector/injector.go

```diff
@@ -0,0 +1,204 @@
+package injector
+
+import (
+	"bufio"
+	"bytes"
+	"github.com/pkg/errors"
+	"io"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kyamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/kube-openapi/pkg/util/sets"
+	"os"
+	"path/filepath"
+)
+
+type (
+	Injector interface {
+		// Injects the given sidecar into all valid Deployment Objects and returns the result as a list of unstructured objects.
+		Inject(sidecar v1.Container) ([]*unstructured.Unstructured, error)
+		// Returns a list of namespaces that contain injectable objects.
+		// This can be used to generate other Kuberenetes objects that need to be created in the same namespace.
+		InjectableNamespaces() ([]string, error)
+	}
+	injectorImpl struct {
+		// The list of Kubernetes objects to traverse during injection. This is a list of
+		// unstructured objects because we likely won't know the type of all objects
+		// ahead of time (e.g., when reading multiple objects from a YAML file).
+		objects []*unstructured.Unstructured
+	}
+)
+
+// Constructs a new Injector with Kubernetes objects derived from the given file path.
+func FromYAML(filePath string) (Injector, error) {
+	yamlContent, err := getFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Read the YAML file into a list of unstructured objects.
+	// This is necessary because the YAML file may contain multiple Kubernetes objects.
+	// We only want to inject the sidecar into Deployment objects, but we still need to parse all resources.
+	multidocReader := kyamlutil.NewYAMLReader(bufio.NewReader(bytes.NewReader(yamlContent)))
+
+	var objList []*unstructured.Unstructured
+	for {
+		raw, err := multidocReader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+			return nil, err
```

Comment:
Is the context for all three of these possible errors clear, or should they be wrapped?